yt/yt/server/master/security_server/security_manager.cpp

```diff
@@ -3925,6 +3940,11 @@ class TSecurityManager
         if (group) {
             return false;
         }
+        group = FindGroupByName(name);
```

Comment:
Using `FindGroupByName` relies on `GroupNameMap_` being initialized. It probably always is, but I'm not 100% sure.

`InitBuiltins` is called from three places: `OnAfterSnapshotLoaded`, `SetZeroState`, and `GetBuiltin`. The first one does initialize `GroupNameMap_` before calling `InitBuiltins`. The second one has no need for that, seeing as it starts from a blank slate. The third one, `GetBuiltin`, however, is trickier. It's called from multiple places and is obviously written for being called very early.

(What I'm afraid of, specifically, is that `GetBuiltin` may be called during snapshot loading - after `GroupMap_` has been loaded by *before* `OnAfterSnapshotLoaded` has been called. If so, the new group will get created and then we'll crash [somewhere around here](https://github.com/ytsaurus/ytsaurus/blob/a4069672b3612fa1676573514413d0afb01bf7b8/yt/yt/server/master/security_server/security_manager.cpp#L3204).)

I don't see an easy way out of this. Some possible options:

  1. Getting rid of `GetBuiltin` - or, at least, the call to `InitBuiltins` inside it. May be trivial. May be impossible. Probably the way to go (or, at least, to attempt to go) before trying out other options.
  2. Moving initialization of `GroupNameMap_` out of `OnAfterSnapshotLoaded` and into a helper method. (And probably also protecting the container with a flag to avoid repeated initialization.) This will entangle dependencies even more. No a pleasant change overall.
  3. Abandoning the idea of checking for pre-existing group (by name) and coding up a full migration from the old "admins" group to the new one. This is doable but not trivial and will require replacing every occurrence of the old group in every ACL and who knows where else.